src/ParameterTable.ts

```diff
@@ -272,4 +252,230 @@ export class ParameterTable {
             //doing it this way because it makes it simpler to have the header in quetion in hand. the ko events proved difficult to pass events and objects with
             upresizer.on('mousedown', mouseDownHandler);
     }
-}
\ No newline at end of file
+}
+
+let localStorageUpdateCooldown : boolean = false;
+
+export class ColumnVisibilities {
+
+    private uiModeName : string;
+    private keyAttribute:ko.Observable<boolean>
+    private displayText:ko.Observable<boolean>
+    private value:ko.Observable<boolean>
+    private readOnly:ko.Observable<boolean>
+    private defaultValue:ko.Observable<boolean>
+    private description:ko.Observable<boolean>
+    private type:ko.Observable<boolean>
+    private parameterType:ko.Observable<boolean>
+    private usage:ko.Observable<boolean>
+    private flags:ko.Observable<boolean>
+    private actions:ko.Observable<boolean>
+
+    constructor(uiModeName:string, keyAttribute:boolean, displayText:boolean,value:boolean,readOnly:boolean,defaultValue:boolean,description:boolean,type:boolean,parameterType:boolean,usage:boolean,flags:boolean,actions:boolean){
+
+        this.uiModeName = uiModeName;
+        this.keyAttribute = ko.observable(keyAttribute);
+        this.displayText = ko.observable(displayText);
+        this.value = ko.observable(value);
+        this.readOnly = ko.observable(readOnly);
+        this.defaultValue = ko.observable(defaultValue);
+        this.description = ko.observable(description);
+        this.type = ko.observable(type);
+        this.parameterType = ko.observable(parameterType);
+        this.usage = ko.observable(usage);
+        this.flags = ko.observable(flags);
+        this.actions = ko.observable(actions);
+
+    }
+
+    getVisibilities = () : ColumnVisibilities => {
+        return this;
+    }
+
+    getModeName = () : string => {
+        return this.uiModeName;
+    }
+
+    getModeByName = (name:string) : ColumnVisibilities => {
+        let columnVisibilityResult:ColumnVisibilities = null
+        columnVisibilities.forEach(function(columnVisibility){
+            if(columnVisibility.getModeName() === name){
+                columnVisibilityResult = columnVisibility
+            }
+        })
+        return columnVisibilityResult
+    }
+
+    setModeName = (newUiModeName:string) : void => {
+        this.uiModeName = newUiModeName;
+    }
+
+    getKeyAttribute = () : boolean => {
+        return this.keyAttribute()
+    }
+
+    private setKeyAttribute = (value:boolean) : void => {
+        if(value === null){
+            this.keyAttribute(!this.keyAttribute());
+            this.saveToLocalStorage()
```

Comment:
this is because this function  is a combination of set and toggle. toggle happens when the provided value is null. this toggle function is used only for the user facing ui, which is where we want to save to local storage. the other part was for set up only.

we talked about this and decided i should separate these combined functions set and toggle. thats done